### PR TITLE
adds doc detailing how declared licenses are determined

### PR DESCRIPTION
**Pull request by nellshamrell**
[Rendered](https://github.com/clearlydefined/service/blob/add-docs-for-declared-licenses/docs/determining-declared-license.md)

Signed-off-by: Nell Shamrell <nells@microsoft.com>

### CODE REVIEW

**Review by disulliv — COMMENTED** [review]
Should this also include information about curations?

**Comment by ariel11** [comment]
> Should this also include information about curations?

@disulliv - are you thinking we should add a link to the [human curation process](https://github.com/clearlydefined/clearlydefined/blob/master/docs/curation-guidelines.md)?

**Comment by nellshamrell** [comment]
I can certainly include a mention that curations would affect the declared license - I'll add that in!

File: docs/determining-declared-license.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+### maven
+* sources https://mvnrepository.com/repos/central and https://maven.google.com/
+
+**maven central**
+
+* The crawler downloads the maven artifact from maven.org
+* The crawler then extracts all pom files from the artifact
+* And then merges the poms (including all the licenses defined in the poms)
+* The ClearlyDefind summarizer parses the merged poms and sets the declared license(s) based on that data
```
Comment (waynebeaton) [inline docs/determining-declared-license.md:43]: s/ClearlyDefind/ClearlyDefined

File: docs/determining-declared-license.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+* It then pulls the registry information for the particular component from that package map file [Code](https://github.com/clearlydefined/crawler/blob/f461b2358fbde130bcc5d183de01a4212c4cd66d/providers/fetch/debianFetch.js#L114)
+* It then finds the relevant copyright URL from the registry information [Code](https://github.com/clearlydefined/crawler/blob/f461b2358fbde130bcc5d183de01a4212c4cd66d/providers/fetch/debianFetch.js#L295) [Example](https://metadata.ftp-master.debian.org/changelogs/main/0/0ad-data/0ad-data_0.0.17-1_copyright)
+* It then pulls information from the copyright URL [Code](https://github.com/clearlydefined/crawler/blob/f461b2358fbde130bcc5d183de01a4212c4cd66d/providers/fetch/debianFetch.js#L306)
+* And parses that information [Code](https://github.com/clearlydefined/crawler/blob/f461b2358fbde130bcc5d183de01a4212c4cd66d/providers/fetch/debianFetch.js#L320) to determine the declared license(s)
+* The ClearlyDefined summarizer then parses the declared licenses and, if necessary, joins them with 'AND'
```
Comment (waynebeaton) [inline docs/determining-declared-license.md:87]: Does "if necessary" mean "if there is more than one"?

**Review by waynebeaton — COMMENTED** [review]
I only managed a quick review. Here are a couple of comments...

**Comment by disulliv** [comment]
> > Should this also include information about curations?
> 
> @disulliv - are you thinking we should add a link to the [human curation process](https://github.com/clearlydefined/clearlydefined/blob/master/docs/curation-guidelines.md)?

I meant more of how curations can affect the declared license, which does involve the human curation process. I think mentioning that curations help determine the declared license (like Nell said) would be good, and including a link to the [human curation process](https://github.com/clearlydefined/clearlydefined/blob/master/docs/curation-guidelines.md) would solidify that understanding.

File: docs/determining-declared-license.md (hunk 2 above)
Comment (nellshamrell) [reply docs/determining-declared-license.md:87]: That is correct - adding that in. Good catch!

**Comment by nellshamrell** [comment]
Thanks @waynebeaton and @disulliv, I have addressed your comments :)

@ariel11 is this the kind of documentation that would be useful to you?

**Review by disulliv — APPROVED** [review]

**Pull request merged**